backend/fms_core/filters.py

```diff
@@ -193,6 +197,21 @@ class Meta:
         model = Dataset
         fields = _dataset_filterset_fields
 
+class ExperimentRunFilter(GenericFilter):
+    experiment_run_progress_stage = django_filters.CharFilter(method="experiment_run_progress_stage_filter")
+
+    def experiment_run_progress_stage_filter(self, queryset, name, value):
+        if value == "processed":
+            return queryset.filter(datasets__readsets__validation_status__in=[ValidationStatus.PASSED.value,ValidationStatus.FAILED.value])
+        if value == "validated":
+            return queryset.filter(datasets__readsets__release_status=ReleaseStatus.AVAILABLE.value,datasets__readsets__validation_status__in=[ValidationStatus.PASSED.value,ValidationStatus.FAILED.value])
```

Comment:
you can use Q queries with queryset.exists(...) statements for the AVAILABLE test. Those are pretty fasts and are easy to read...